ext/sdk/resources/sdk-root/shell/src/resource-templates/ts/scaffolder.ts

```diff
@@ -0,0 +1,212 @@
+import { inject, injectable } from "inversify";
+import { FsService } from "backend/fs/fs-service";
+import { concurrently } from "utils/concurrently";
+import { ResourceTemplateScaffolder, ResourceTemplateScaffolderArgs } from "../types";
+
+@injectable()
+export default class TsScaffolder implements ResourceTemplateScaffolder {
+  @inject(FsService)
+  protected readonly fsService: FsService;
+
+  async scaffold({ request, manifest, resourcePath }: ResourceTemplateScaffolderArgs) {
+    const resourceName = request.assetName;
+    
+    manifest.clientScripts.push('dist/*.client.js');
+    manifest.serverScripts.push('dist/*.server.js');
+    
+    const promises = [];
+    const relativePath = (to: string) => this.fsService.joinPath(resourcePath, to);
+
+
+    // webpack and package.json
+    promises.push(this.fsService.writeFile(relativePath('webpack.config.js'), getWebpackContent()));
+    promises.push(this.fsService.writeFile(relativePath('package.json'), getPackageContent(resourceName))); 
+    
+    // client
+    promises.push(
+      this.fsService.mkdirp(relativePath('client'))
+        .then(() => concurrently(
+          this.fsService.writeFile(relativePath('client/client.ts'), ''),
+          this.fsService.writeFile(relativePath('client/tsconfig.json'), getTsConfigClient())
+        )),
+    );
+
+    // server
+    promises.push(
+      this.fsService.mkdirp(relativePath('server'))
+        .then(() => concurrently(
+          this.fsService.writeFile(relativePath('server/server.ts'), ''),
+          this.fsService.writeFile(relativePath('server/tsconfig.json'), getTsConfigServer())
+        )),
+    );
+
+
+    await Promise.all(promises);
+  };
+}
+
+function getTsConfigClient() {
+  return `
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "*": ["types/*"]
+    },
+    "outDir": "./",
+    "noImplicitAny": true,
+    "module": "commonjs",
+    "target": "es6",
+    "allowJs": true,
+    "lib": ["es2017"],
+    "types": ["@citizenfx/client", "@types/node"],
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "esModuleInterop": true
+  },
+  "include": ["./**/*"],
+  "exclude": ["**/node_modules"]
+}
+  `
+}
+
+
+function getTsConfigServer() {
+  return `
+  {
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "*": ["types/*"]
+    },
+    "outDir": "./",
+    "noImplicitAny": true,
+    "module": "commonjs",
+    "target": "es6",
+    "allowJs": false,
+    "lib": ["es2015", "DOM"],
+    "types": ["@citizenfx/server", "@types/node"],
+    "moduleResolution": "node",
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
```

Comment:
Kinda pretty specific need, why would this be in template?

Reply:
Are you talking about just line 89 -> 91?

I agree, but I think 88 makes sense? It's just the proper types for the server section

Reply:
> Are you talking about just line 89 -> 91?

90 and 91.